csharp/ql/src/semmle/code/csharp/commons/Util.qll

```diff
@@ -13,3 +13,13 @@ class MainMethod extends Method {
     else this.getNumberOfParameters() = 0
   }
 }
+
+class RecordCloneMethod extends Method {
+  RecordCloneMethod() { exists(WithExpr we | we.getTarget() = this) }
+
+  Constructor getConstructor() {
+    this.getDeclaringType().hasMember(result) and
+    result.getNumberOfParameters() = 1 and
+    result.getParameter(0).getType() = this.getDeclaringType()
+  }
```

Comment:
If the `record` is `abstract`, we have no such constructor. Maybe worth making it more explicit.